docs/configuration/configuration.md

```diff
@@ -46,6 +46,7 @@ An example [oauth2-proxy.cfg]({{ site.gitweb }}/contrib/oauth2-proxy.cfg.example
 | `--display-htpasswd-form` | bool | display username / password login form if an htpasswd file is provided | true |
 | `--email-domain` | string \| list  | authenticate emails with the specified domain (may be given multiple times). Use `*` to authenticate any email | |
 | `--extra-jwt-issuers` | string | if `--skip-jwt-bearer-tokens` is set, a list of extra JWT `issuer=audience` pairs (where the issuer URL has a `.well-known/openid-configuration` or a `.well-known/jwks.json`) | |
+| `--extra-header` | string \| list | Add additional headers when talking to upstream in raw http header format. eg: `"Foo=Bar"` | |
```

Comment:
An example for multiple values maybe?
```suggestion
| `--extra-header` | string \| list | Add additional headers when talking to upstream in raw http header format. eg: `"Foo=Bar"` or `"Bar=Foo,Baz"` for multiple values | |
```

Reply:
Should this flag be named `--pass-extra-headers` to align with the `pass` prefix that other upstream header setting options use?

(Out of scope for this PR, just sharing the idea): This gives us the flexibility to add on to this functionality and potentially implement a similar `set-extra-headers` to handle setting extra headers on the response headers to the client?

I could see this potentially being useful for cases where people use nginx `auth_request` like functionality in a kubernetes ingress environment with ingress auth annotations alone -- so they don't have access to set headers directly in an nginx config. (or via annotation they point at different oauth2-proxy deployments, so those can potentially set different response headers)

Reply:
@kfox1111 Can you also rename this flag to `pass-extra-headers` to align to the naming convention we have for all other upstream header passing flags?

Reply:
The problem with `pass-extra-headers` is that the `s` on the end breaks the multiple flag specification rules that we have, it should be singular I think